editor/editor_notes.cpp

```diff
@@ -159,16 +167,10 @@ void Notes::Upload(osm::OsmOAuth const & auth)
   auto const self = shared_from_this();
 
   auto const doUpload = [self, auth]() {
-    size_t size;
-
-    {
-      lock_guard<mutex> g(self->m_mu);
-      size = self->m_notes.size();
-    }
+    lock_guard<mutex> g(self->m_mu);
 
     osm::ServerApi06 api(auth);
-    auto it = begin(self->m_notes);
-    for (size_t i = 0; i != size; ++i, ++it)
+    for (auto it = self->m_notes.begin(); it != self->m_notes.end(); ++it)
```

Comment:
We skip some notes here:
```it = self->m_notes.erase(it);``` returns iterator to an element that follows the one which was removed.
At the end of the iteration ```for (auto it = self->m_notes.begin(); it != self->m_notes.end(); ++it)```
moves iterator even further. So we always skip an element that follows current iterator.